applications/CompressiblePotentialFlowApplication/custom_processes/define_3d_wake_process.cpp

```diff
@@ -0,0 +1,906 @@
+#include "processes/process.h"
+#include "compressible_potential_flow_application_variables.h"
+#include "define_3d_wake_process.h"
+#include "utilities/variable_utils.h"
+#include "custom_utilities/potential_flow_utilities.h"
+#include "processes/calculate_distance_to_skin_process.h"
+#include "utilities/builtin_timer.h"
+
+namespace Kratos {
+
+// Constructor for Define3DWakeProcess Process
+Define3DWakeProcess::Define3DWakeProcess(ModelPart& rTrailingEdgeModelPart,
+                                         ModelPart& rBodyModelPart,
+                                         ModelPart& rStlWakeModelPart,
+                                         Parameters ThisParameters)
+    : Process(),
+      mrTrailingEdgeModelPart(rTrailingEdgeModelPart),
+      mrBodyModelPart(rBodyModelPart),
+      mrStlWakeModelPart(rStlWakeModelPart)
+{
+    Parameters default_parameters = Parameters(R"(
+    {
+        "tolerance"                            : 1e-9,
+        "wake_normal"                          : [0.0,0.0,1.0],
+        "wake_direction"                       : [1.0,0.0,0.0],
+        "switch_wake_normal"                   : false,
+        "count_elements_number"                : false,
+        "write_elements_ids_to_file"           : false,
+        "shed_wake_from_trailing_edge"         : false,
+        "shedded_wake_distance"                : 12.5,
+        "shedded_wake_element_size"            : 0.2,
+        "decrease_wake_width_at_the_wing_tips" : false,
+        "echo_level": 1
+    })" );
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    mTolerance = ThisParameters["tolerance"].GetDouble();
+    mWakeNormal = ThisParameters["wake_normal"].GetVector();
+    mWakeDirection = ThisParameters["wake_direction"].GetVector();
+    mSwitchWakeDirection = ThisParameters["switch_wake_normal"].GetBool();
+    mCountElementsNumber = ThisParameters["count_elements_number"].GetBool();
+    mWriteElementsIdsToFile = ThisParameters["write_elements_ids_to_file"].GetBool();
+    mShedWakeFromTrailingEdge = ThisParameters["shed_wake_from_trailing_edge"].GetBool();
+    mSheddedWakeDistance = ThisParameters["shedded_wake_distance"].GetDouble();
+    mSheddedWakeElementSize = ThisParameters["shedded_wake_element_size"].GetDouble();
+    mDecreaseWakeWidthAtTheWingTips = ThisParameters["decrease_wake_width_at_the_wing_tips"].GetBool();
+    mEchoLevel = ThisParameters["echo_level"].GetInt();
+
+    KRATOS_ERROR_IF(mWakeNormal.size() != 3)
+        << "The mWakeNormal should be a vector with 3 components!"
+        << std::endl;
+}
+
+void Define3DWakeProcess::ExecuteInitialize()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    auto& r_nodes = root_model_part.Nodes();
+    VariableUtils().SetNonHistoricalVariable(UPPER_SURFACE, false, r_nodes);
+    VariableUtils().SetNonHistoricalVariable(LOWER_SURFACE, false, r_nodes);
+    VariableUtils().SetNonHistoricalVariable(TRAILING_EDGE, false, r_nodes);
```

Comment:
I would put all of them in one `block_for_each` rather than calling `VariableUtils` thrice. This make looping over all the nodes three times.

Reply:
makes sense, I'll do it!